src/components/Topics.jsx

```diff
@@ -1,37 +1,67 @@
 import { useEffect, useState } from "react";
 import { useParams } from "react-router-dom"
-import { getArticles,getArticlesSorted } from "../../api"
+import { getArticles,getArticlesSorted, getTopics } from "../../api"
 import SingleArticle from "./SingleArticle";
+import TopicList from "./TopicsList";
 
 
 
 
 const Topic = ()=>{
 const topic = useParams();
+
+
 const [isLoading, setIsLoading] =useState(true)
 const [articles, setArticles] =useState([])
 const[error, setError] = useState(false)
 const [order, setOrder] = useState("desc")
+const [topics, setTopics] = useState([])
+const [apiError, setApiError] = useState(false)
+
 
     useEffect(()=>{
         setIsLoading(true)
         setError(true)
 
 
         getArticles()
+
         .then((res)=>{
 
             setArticles(res)
             setIsLoading(false)
             setError(false)
         })
         .catch(err =>{
+
             setIsLoading(false)
             setError(true)
     
         })
     },[])
 
+    useEffect(()=>{
```

Comment:
I wonder if there is a way to only use one useEffect here - it doesn't feel very DRY.